src/sqlfluff/core/rules/base.py

```diff
@@ -798,20 +798,26 @@ def _choose_anchor_segment(context, edit_type, segment, filter_meta=False):
         anchor = segment
         child = segment
         for seg in context.parent_stack[0].path_to(segment)[1:-1][::-1]:
-            children = (
-                seg.segments
-                if not filter_meta
-                else [child for child in seg.segments if not child.is_meta]
-            )
-            if edit_type == "create_before" and children[0] is child:
-                anchor = seg
-                assert anchor.raw.startswith(segment.raw)
-            elif edit_type == "create_after" and children[-1] is child:
-                anchor = seg
-                assert anchor.raw.endswith(segment.raw)
-            else:
-                break
-            child = seg
+            # Which sets of children to check against.
```

Comment:
Why did this all need to change? As per your description I would have thought the first set of changes should be enough?

Reply:
The `_choose_anchor_segment()` changes fixed an issue with L003. Once I fixed that, I was getting a second parse issue with L052, where it was placing the trailing semicolon too low in the tree. Fixing this allowed the semicolon to be placed correctly. I'll see if I can add a test for that.

Reply:
Can we also add a comment to the PR description then to note that's include in this PR too then?

Reply:
I can't reproduce the L052 issue in isolation. If you think it's worthwhile, I can add a test that runs both L003 and L052 -- that would serve as a test for this change. Let me know!

Reply:
Well this code is covered. But still weird tests are good to have so maybe should add?

Reply:
Test added!